packages/components/src/color-palette/index.js

```diff
@@ -143,7 +146,14 @@ export default function ColorPalette( {
 							aria-haspopup="true"
 							onClick={ onToggle }
 							aria-label={ __( 'Custom color picker' ) }
-							style={ { background: value } }
+							style={ {
+								background: value,
+								color:
+									colordColor.contrast() >
+									colordColor.contrast( '#000' )
+										? '#fff'
+										: '#000',
```

Comment:
I realize this snippet is copy & pasted from the same file, but I found that we actually have a nice [utility function](https://github.com/WordPress/gutenberg/blob/406db6277bf965fdcaa7db1d2d84bbbee1a0f697/packages/block-editor/src/components/colors/utils.js#L78) for this.